tsl/src/bgw_policy/job.c

```diff
@@ -250,9 +250,50 @@ bool
 policy_retention_execute(int32 job_id, Jsonb *config)
 {
 	PolicyRetentionData policy_data;
+	char *relname;
+	char *message = "applying retention policy to hypertable";
+	Datum boundary;
 
 	policy_retention_read_and_validate_config(config, &policy_data);
 
+	relname = get_rel_name(policy_data.object_relid);
+	boundary = policy_data.boundary;
+	if (IS_INTEGER_TYPE(policy_data.boundary_type))
+	{
+		elog(LOG,
+			 "%s: %s dropping data older than %s",
+			 message,
+			 relname,
+			 DatumGetCString(DirectFunctionCall1(int8out, boundary)));
+	}
+	switch (policy_data.boundary_type)
+	{
```

Comment:
This whole `if`, `switch()` block should be in a separate function that returns a boundary string. Also, the `if` and the `switch` should not both be executed. Maybe add an `else` to the `switch`.